tests/forms_tests/tests/test_formsets.py

```diff
@@ -1481,6 +1481,16 @@ def test_disable_delete_extra_formset_forms(self):
         self.assertNotIn("DELETE", formset.forms[1].fields)
         self.assertNotIn("DELETE", formset.forms[2].fields)
 
+        formset = ChoiceFormFormset(initial=[{"choice": "Zero", "votes": "1"}])
+        forms = {}
+        for index in [0, 1, None]:
+            form = formset.form()
+            formset.add_fields(form, index)
+            forms[index] = form
+        self.assertIn("DELETE", forms[0].fields)
+        self.assertNotIn("DELETE", forms[1].fields)
+        self.assertNotIn("DELETE", forms[None].fields)
+
```

Comment:
I'd test `empty_form` instead of direct testing of `add_fields()`:
```suggestion
        self.assertNotIn("DELETE", formset.empty_form.fields)
```